pkg/limits/ingest_limits.go

```diff
@@ -350,43 +319,6 @@ func (s *IngestLimits) running(ctx context.Context) error {
 	}
 }
 
-// evictOldStreams evicts old streams. A stream is evicted if it has not
-// been seen within the window size.
-func (s *IngestLimits) evictOldStreams(_ context.Context) {
-	s.mtx.Lock()
-	defer s.mtx.Unlock()
-
-	cutoff := time.Now().Add(-s.cfg.WindowSize).UnixNano()
-
-	for tenant, partitions := range s.metadata {
-		evicted := 0
-		for partitionID, streams := range partitions {
-			// Create a new slice with only active streams
-			activeStreams := make([]streamMetadata, 0, len(streams))
-			for _, stream := range streams {
-				if stream.lastSeenAt >= cutoff {
-					activeStreams = append(activeStreams, stream)
-				} else {
-					evicted++
-				}
-			}
-			s.metadata[tenant][partitionID] = activeStreams
-
-			// If no active streams in this partition, delete it
-			if len(activeStreams) == 0 {
-				delete(s.metadata[tenant], partitionID)
-			}
-		}
-
-		// If no partitions left for this tenant, delete the tenant
-		if len(s.metadata[tenant]) == 0 {
-			delete(s.metadata, tenant)
-		}
-
-		s.metrics.tenantStreamEvictionsTotal.WithLabelValues(tenant).Add(float64(evicted))
-	}
-}
-
 // evictOldStreamsPeriodic runs a periodic job that evicts old streams.
 // It runs two evictions per window size.
 func (s *IngestLimits) evictOldStreamsPeriodic(ctx context.Context) {
```

Comment:
We can use the evictor I added in `evictor.go` as a follow up (doesn't need to happen in this PR). Might use it for the cache too.